src/std.rs

```diff
@@ -1,20 +1,32 @@
 //! libstd-based implementation of `event-listener`.
 //!
-//! This implementation crates an intrusive linked list of listeners.
+//! This implementation crates an intrusive linked list of listeners. This list
+//! is secured using either a libstd mutex or a critical section.
 
 use crate::notify::{GenericNotify, Internal, Notification};
 use crate::sync::atomic::Ordering;
 use crate::sync::cell::{Cell, UnsafeCell};
-use crate::sync::{Mutex, MutexGuard};
 use crate::{RegisterResult, State, TaskRef};
 
+#[cfg(feature = "critical-section")]
+use core::cell::RefCell;
```

Comment:
nitpick: I have learnt that it's better to not import conditionally as it then increases the `cfg` lines. E.g I personally favour

```rust
#[cfg(features = "xyz")]
xyz::Xyz::foo();

...

#[cfg(features = "xyz")]
xyz::Xyz::bar();
```

over

```rust
#[cfg(features = "xyz")]
use xyz::Xyz;

...

#[cfg(features = "xyz")]
Xyz::blah();

...

#[cfg(features = "xyz")]
Xyz::bar();
```

Of course it's a personal taste so up to you. :)

Reply:
I prefer to avoid using full paths in the body of the code.

Reply:
normally, so do I. It's just that the code becomes much harder to follow if imports are conditional. Anyway, as I wrote, up to you. :)